plugins/output/adapt/helpers.js

```diff
@@ -0,0 +1,276 @@
+* @param {string} filePath
+* @param {string} unzipPath
+* @param {callback} done
+*/
+function unzip(filePath, unzipPath, done) {
+  yauzl.open(filePath, { lazyEntries: true }, function(error, zipfile) {
+    if (error) {
+      return done(error);
+    }
+    zipfile.readEntry();
+    zipfile.on("entry", function(entry) {
+      var dest = path.join(unzipPath, entry.fileName);
+      if (/\/$/.test(entry.fileName)) {
+        fs.ensureDir(dest, function(err) {
+          if (error) {
+            return done(error);
+          }
+          zipfile.readEntry();
+        });
+      } else {
+        zipfile.openReadStream(entry, function(err, readStream) {
+          if (error) {
+            return done(error);
+          }
+          fs.ensureDir(path.dirname(dest), function(err) {
+            if (error) {
+              return done(error);
+            }
+            readStream.pipe(fs.createWriteStream(dest));
+            readStream.on("end", function() {
+              zipfile.readEntry();
+            });
+          });
+        });
+      }
+    });
+    zipfile.once('end', done);
+  });
+}
+
+
+/**
+* import a single plugin
+* @param {string} pluginDir
+* @param {string} pluginType
+* @param {callback} pluginImported
+*/
+function importPlugin(pluginDir, pluginType, pluginImported) {
+  var bowerJson, contentPlugin, frameworkVersion;
+
+  async.waterfall([
+    function readBowerJson(cb) {
+      fs.stat(path.join(pluginDir, Constants.Filenames.Bower), function(error, stats) {
+        if (error) {
+            logger.log('error', error)
+            return pluginImported();
+        }
+        fs.readJson(path.join(pluginDir, Constants.Filenames.Bower), cb);
+      });
+    },
+    function getContentPlugin(json, cb) {
+      bowerJson = json;
+      origin.contentmanager.getContentPlugin(pluginType, cb);
+    },
+    function getDB(plugin, cb) {
+      contentPlugin = plugin;
+      database.getDatabase(cb);
+    },
+    function retrievePluginDoc(db, cb) {
+      db.retrieve(contentPlugin.bowerConfig.type, { name: bowerJson.name }, { jsonOnly: true }, cb);
+    },
+    function addPlugin(records, cb) {
+      if(records.length === 0) {
+        installHelpers.getInstalledFrameworkVersion(function(err, frameworkVersion ) {
+          if(err){
+            return cb(err)
+          }
+
+          var serverVersion = semver.clean(frameworkVersion);
+          var pluginRange = semver.validRange(bowerJson.framework);
+
+          if(semver.satisfies(serverVersion, pluginRange)) {
+            logger.log('info', 'Installing', pluginType, "'" + bowerJson.displayName + "'");
+            bowerJson.isLocalPackage = true;
+            app.bowermanager.addPackage(contentPlugin.bowerConfig, { canonicalDir: pluginDir, pkgMeta: bowerJson }, { strict: true }, cb);
+          } else {
+            logger.log('info', "Can't install " + bowerJson.displayName + ", it requires framework v" + pluginVersion + " (" + frameworkVersion + " installed)");
```

Comment:
Getting a `ReferenceError` related to `pluginVersion` here. Should this be `pluginRange` instead?

Reply:
That is correct, just changed 👍